src/marketlist.schema.json

```
@@ -129,18 +128,15 @@
           "pattern": "^0x[a-fA-F0-9]{40}$"
         },
         "signature": {
-          "enum": [
-            "function invest(uint256[] calldata amounts) external payable",
-            "function invest(uint256[] calldata amounts, bytes calldata data) external payable"
-          ],
+          "pattern": "^function\\sinvest((s*([a-zA-Z_]w+)s+([a-zA-Z_]w+),?)*)",
           "type": "string"
         }
       },
       "required": ["address", "signature"]
     },
     "InvestmentOpportunity": {
       "type": "object",
-      "description": "Properties required for detailing how a user can enter a protected investment after borrowing from the protection market",
+      "description": "Properties required for detailing how a user can enter a protected investment after borrowing from the market",
```

nit: not sure which is more precise 
```suggestion
      "description": "Properties required for detailing how a user can enter a protected investment after borrowing from a market",
```

---

I don't care particularly, but since you're defining the investment opportunity in the context of a market, "the" seems a bit better to me.